File: apps/api/app/schemas/trust.py
"""
Pydantic schemas for trust-and-compliance surfaces:

  * "Canlı Veri" badge — per-symbol live data provenance
  * Transparency page — full provider list + policy links
  * Investment-advice disclaimer — localized, signed payload
  * "Çelik Hesap" badge — security posture scoring
"""

from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel


class DataSourceBadge(BaseModel):
    symbol: str
    source_label: str
    source_code: str
    last_updated_at: datetime | None
    age_seconds: int | None
    freshness: Literal["live", "recent", "stale"]
    badge_tone: Literal["positive", "neutral", "warning"]
    disclosure: str


class ProviderEntry(BaseModel):
    code: str
    name: str
    category: Literal["market_data", "macro", "fx", "commodity", "news", "security"]
    description: str
    coverage: list[str]
    website_url: str | None
    terms_url: str | None


class TransparencyView(BaseModel):
    providers: list[ProviderEntry]
    policies: list[dict]
    last_reviewed_at: datetime


class DisclaimerView(BaseModel):
    locale: str
    title: str
    body: str
    acknowledgement_cta: str
    version: str
    effective_at: datetime


class SecurityControl(BaseModel):
    code: Literal[
        "email_verified",
        "two_factor_totp",
        "biometric_app_lock",
        "strong_password",
        "recovery_email",
        "push_confirmation",
    ]
    label: str
    enabled: bool
    weight: int
    tip: str | None


class SteelAccountView(BaseModel):
    score: int
    max_score: int
    tier: Literal["starter", "shielded", "steel", "titanium"]
    is_steel: bool
    controls: list[SecurityControl]
    next_action: str | None
    badge_updated_at: datetime

File: apps/mobile/src/types/trust.ts
export interface DataSourceBadge {
  symbol: string;
  source_label: string;
  source_code: string;
  last_updated_at: string | null;
  age_seconds: number | null;
  freshness: 'live' | 'recent' | 'stale';
  badge_tone: 'positive' | 'neutral' | 'warning';
  disclosure: string;
}

export interface ProviderEntry {
  code: string;
  name: string;
  category: 'market_data' | 'macro' | 'fx' | 'commodity' | 'news' | 'security';
  description: string;
  coverage: string[];
  website_url: string | null;
  terms_url: string | null;
}

export interface TransparencyView {
  providers: ProviderEntry[];
  policies: Array<{ code: string; title: string; url: string }>;
  last_reviewed_at: string;
}

export interface DisclaimerView {
  locale: string;
  title: string;
  body: string;
  acknowledgement_cta: string;
  version: string;
  effective_at: string;
}

export interface SecurityControl {
  code:
    | 'email_verified'
    | 'two_factor_totp'
    | 'biometric_app_lock'
    | 'strong_password'
    | 'recovery_email'
    | 'push_confirmation';
  label: string;
  enabled: boolean;
  weight: number;
  tip: string | null;
}

export interface SteelAccountView {
  score: number;
  max_score: number;
  tier: 'starter' | 'shielded' | 'steel' | 'titanium';
  is_steel: boolean;
  controls: SecurityControl[];
  next_action: string | null;
  badge_updated_at: string;
}
